sql/grant_tables/role_edge.go

```diff
@@ -72,3 +75,25 @@ func (r *RoleEdge) Equals(ctx *sql.Context, otherEntry in_mem_table.Entry) bool
 	}
 	return *r == *otherRoleEdge
 }
+
+// ToStringWithQuote returns the "TO" user as a formatted string using the quotes given. If a replacement is given, then
+// also replaces any existing instances of the quotes with the replacement.
+func (r *RoleEdge) ToStringWithQuote(quote string, replacement string) string {
+	return r.stringWithQuote(r.ToUser, r.ToHost, quote, replacement)
+}
+
+// FromStringWithQuote returns the "FROM" user as a formatted string using the quotes given. If a replacement is given,
+// then also replaces any existing instances of the quotes with the replacement.
+func (r *RoleEdge) FromStringWithQuote(quote string, replacement string) string {
```

Comment:
Very odd interface here, makes no sense even after looking at the call site. Why would we replace one backtick with two in the host and name identifiers? Why do the identifiers have backticks in them in the first place?

Reply:
`root@localhost` would become `’root’@‘localhost’`. Different quotes are used in different places, so this is mimicking that.

Reply:
In MySQL the proper way to escape a quote is to double it, but it makes more sense to just enforce that rather than optionally provide the replacement as another parameter.